operator/internal/controller/transaction_reconcile.go

```diff
@@ -215,9 +230,40 @@ func getPaladinRPC(ctx context.Context, c client.Client, nodeName, namespace str
 	if err != nil {
 		return nil, err
 	}
-	return pldclient.New().HTTP(ctx, &pldconf.HTTPClientConfig{
+
+	// Adding the timeout to the cache key to ensure that different timeouts are cached separately
+	// This is important because the timeout is used in the HTTP client config
+	// and different timeouts may require different configurations.
+	key := fmt.Sprintf("%s-%s", url, timeout)
+
+	// Check if the client is already in the cache
+	// Use a read lock to avoid blocking other goroutines
+	cache.mux.RLock()
+	if client, ok := cache.clients[key]; ok && client != nil {
+		cache.mux.RUnlock()
+		return client, nil
+	}
+	cache.mux.RUnlock()
+
+	// If not, create a new client and store it in the cache
+	cache.mux.Lock()
+	defer cache.mux.Unlock()
+
+	// Check again in the cache after acquiring the lock
+	// This is to ensure that another goroutine didn't create the client while we were waiting for the lock
+	if client, ok := cache.clients[key]; ok && client != nil {
+		return client, nil
+	}
+
+	client, err := pldclient.New().HTTP(ctx, &pldconf.HTTPClientConfig{
```

Comment:
I guess - the operator creating a new client/transport every reconcile is really inefficient but that doesn't fully explain the memory pressure right ?

There must be something holding a reference in memory to this client after the reconcile loop that keeps it around ? Is that a bug in `pldclient` itself ? Bc that seems odd.

Still - a client manager in the operator to track clients per Paladin's is a great/defensive optimization either way.

Reply:
I mentioned this bc in ff-common we had a really bad bug in the client library itself that caused a memory leak: https://github.com/hyperledger/firefly-common/pull/172